lisp/x/profiler/go_annotator.go

```diff
@@ -0,0 +1,98 @@
+package profiler
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/golang-collections/collections/stack"
+	"github.com/luthersystems/elps/lisp"
+	"runtime/pprof"
+	"time"
+)
+
+// This profiler type appends tags to pprof output if pprof is enabled.
+// I've taken the decision to not start pprof for the user as I suspect that would be more
+// annoying than useful in the majority of contexts where this could be used.
+// Unfortunately, due to the decision of the writers of pprof to fix the sampling rate at 100Hz
+// this is now less useful than might be expected.
+type pprofAnnotator struct {
+	runtime        *lisp.Runtime
+	enabled        bool
+	startTime      time.Time
+	currentContext context.Context
+	contexts       *stack.Stack
+}
+
+func NewPprofAnnotator(runtime *lisp.Runtime, parentContext context.Context) lisp.Profiler {
```

Comment:
It's happy because it's a `New...` function, I think, and the parameter isn't called `ctx`. I'd deliberately made the decision to have it as the second parameter to make clear this isn't conventional golang context passing.